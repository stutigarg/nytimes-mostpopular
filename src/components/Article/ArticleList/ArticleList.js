import React from 'react';
import ArticleItem from "./ArticleItem";

const ArticleList = ({articles, onSelectHandler}) =>{
        const renderedArticles =  articles.map((article) => {
            return (
                <ArticleItem
                    item={article}
                    key={article.id}
                    onSelectHandler={onSelectHandler}
                />
            )
        });
        return  <ul className="list-unstyled">{renderedArticles}</ul>;
    }

export default ArticleList;