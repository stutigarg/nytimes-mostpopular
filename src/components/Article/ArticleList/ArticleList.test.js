import React from 'react';
import {configure, shallow} from 'enzyme';
import ArticleList from './ArticleList';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const articleOne = {
    id: 1,
    title: 'A really interesting article.'
};

const articleTwo = {
    id: 2,
    title: 'Another interesting article.'
};

const articles = [articleOne, articleTwo];

describe('<ArticleList />', () => {
  test('renders', () => {
    const wrapper = shallow(<ArticleList articles={articles} />);
    expect(wrapper).toMatchSnapshot();
  });
});
