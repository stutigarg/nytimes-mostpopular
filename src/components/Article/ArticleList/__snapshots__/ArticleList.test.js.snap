// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleList /> renders 1`] = `ShallowWrapper {}`;
